src/main.rs

```diff
@@ -72,33 +80,44 @@ async fn main() -> Result<(), APIError> {
         _ => loader.download_model(model_id, None, args.hf_token, args.hf_token_path)?,
     };
 
-    let dtype = match args.dtype.as_deref() {
-        Some("f16") => DType::F16,
-        Some("bf16") => DType::BF16,
-        Some("f32") => DType::F32,
+    let (dtype, dsize) = match args.dtype.as_deref() {
```

Comment:
Can we use `DType::size_in_bytes` instead of the hardcoded values here for readability?

Reply:
> Can we use `DType::size_in_bytes` instead of the hardcoded values here for readability?

Yes, sure. :)